app/models/quote.rb

```diff
@@ -1,25 +1,50 @@
 class Quote < ActiveRecord::Base
   has_paper_trail ignore: %i[text]
   acts_as_paranoid
-  belongs_to :user
-  belongs_to :reporter, class_name: "User"
-  validates :user, presence: true
-  validates :reporter, presence: true
+  belongs_to :user, optional: true
+  belongs_to :reporter, optional: true, class_name: "User"
   validates :text, presence: true
   serialize :text
   validate :reporter_and_user_differ
+  after_save :destroy_versions if :anonymous?
 
   encrypts :text
 
   scope :ordered, -> { order('created_at DESC') }
   scope :random, -> { order('RAND()') }
   scope :with_user, -> { includes(:user) }
+  scope :anonymous, -> { where(user_id: nil) }
+  scope :not_anonymous, -> { where.not(user_id: nil) }
+
+  def anonymous?
+    user.nil?
+  end
+
+  def can_be_anonymized_by?(user)
+    !anonymous? && user.admin? || user.id == user_id
```

Comment:
If you are the user then you can anonymise if it is already anonymous? The first check should be a guard probably

Reply:
Goed punt, stuk netter